plugins/devtools-backend/src/api/DevToolsBackendApi.ts

```diff
@@ -0,0 +1,222 @@
+          result.push(pingResult);
+          break;
+        }
+        case 'fetch': {
+          const fetchResult = await this.fetchExternalDependency(endpoint);
+          result.push(fetchResult);
+          break;
+        }
+        default:
+          return result;
+      }
+    }
+
+    return result;
+  }
+
+  private async fetchExternalDependency(
+    endpoint: Endpoint,
+  ): Promise<ExternalDependency> {
+    let status;
+    let error;
+
+    await fetch(endpoint.target)
+      .then(res => {
+        status = res.status === 200 ? 'Healthy' : 'Unhealthy';
+        this.logger.info(
+          `Fetch for ${endpoint.name} resulted in status code "${res.status}"`,
+        );
+      })
+      .catch((err: Error) => {
+        this.logger.error(`Fetch failed for ${endpoint.name} - ${err.message}`);
+        error = err.message;
+      });
+
+    const result: ExternalDependency = {
+      name: endpoint.name,
+      type: endpoint.type,
+      target: endpoint.target,
+      status: status ?? 'Unhealthy',
+      error: error ?? undefined,
+    };
+
+    return result;
+  }
+
+  private async pingExternalDependency(
+    endpoint: Endpoint,
+  ): Promise<ExternalDependency> {
+    const pingResult = await ping.promise.probe(endpoint.target);
+
+    let error;
+    if (
+      pingResult.packetLoss === '100.000' ||
+      pingResult.packetLoss === 'unknown'
+    ) {
+      this.logger.error(
+        `Ping failed for ${endpoint.name} - ${pingResult.output}`,
+      );
+      error = pingResult.output === '' ? `${endpoint.target} - Unknown` : pingResult.output;
+    }
+
+    this.logger.info(`Ping results for ${endpoint.name}: ${pingResult.output}`);
+
+    const result: ExternalDependency = {
+      name: endpoint.name,
+      type: endpoint.type,
+      target: endpoint.target,
+      status: pingResult.alive ? 'Healthy' : 'Unhealthy',
+      error: error ?? undefined,
+    };
+
+    return result;
+  }
+
+  public async listConfig(): Promise<JsonValue> {
+    const args = parseArgs(process.argv);
+
+    const configTargets: ConfigTarget[] = [args.config ?? []]
+      .flat()
+      .map(arg =>
+        isValidUrl(arg) ? { url: arg } : { path: resolvePath(arg) },
+      );
+
+    /* eslint-disable-next-line no-restricted-syntax */
+    const paths = findPaths(__dirname);
+
+    const { packages } = await getPackages(paths.targetDir);
+    const schema = await loadConfigSchema({
+      dependencies: packages.map(p => p.packageJson.name),
+    });
+
+    const { appConfigs } = await loadConfig({
+      configRoot: paths.targetRoot,
+      configTargets: configTargets,
+    });
+
+    const safeAppConfigs = schema.process(appConfigs, {
+      visibility: ['frontend', 'backend'],
+      withDeprecatedKeys: true,
+    });
+    const safeConfig = ConfigReader.fromConfigs(safeAppConfigs);
+
+    const data = safeConfig.get();
+
+    return data;
+  }
+
+  public async listInfo(): Promise<DevToolsInfo> {
```

Comment:
Noting that this won't work for split projects where not all packages are in the same workspace. Tbh I don't quite know what we could do about that though other than to try to federate it in the backend and somehow bring the information in in the frontend as well.

There's #10256 brewing as well, which loads in the package.json files of plugins, which you can probably utilize.

Reply:
Yeah, honestly I've been approaching this as a MVP (Minimal Viable Product) so it's not going to be perfect in all cases. My hope, though possibly naive, is that as this gets used the community will help improve it over time.

Yes, #10256 will be worth integrating with once it gets in 👍